buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-							t.Fatalf("Unexpected error:\n%s\n", err)
-						}
-						if s := cmp.Diff(br, buildpack.BuildResult{
-							LaunchBOM:   nil,
-							Labels:      []buildpack.Label{},
-							MetRequires: nil,
-							Processes: []launch.Process{
-								{Type: "some-type", Command: "some-cmd", BuildpackID: "A", Default: true},
-								{Type: "web", Command: "other-cmd", BuildpackID: "A", Default: false},
-							},
-							Slices: []layers.Slice{},
-						}); s != "" {
-							t.Fatalf("Unexpected metadata:\n%s\n", s)
-						}
-					})
-
-					it("should set the working directory", func() {
-						h.Mkfile(t,
-							"[[processes]]\n"+
-								`working-dir = "/working-directory"`,
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-						br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNil(t, err)
-						h.AssertEq(t, len(br.Processes), 1)
-						h.AssertEq(t, br.Processes[0].WorkingDirectory, "/working-directory")
-					})
-				})
-
-				when("slices", func() {
-					it("should include slices", func() {
-						h.Mkfile(t,
-							"[[slices]]\n"+
-								`paths = ["some-path", "some-other-path"]`+"\n",
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-
-						br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						if err != nil {
-							t.Fatalf("Unexpected error:\n%s\n", err)
-						}
-
-						if s := cmp.Diff(br, buildpack.BuildResult{
-							LaunchBOM:   nil,
-							Labels:      []buildpack.Label{},
-							MetRequires: nil,
-							Processes:   []launch.Process{},
-							Slices:      []layers.Slice{{Paths: []string{"some-path", "some-other-path"}}},
-						}); s != "" {
-							t.Fatalf("Unexpected:\n%s\n", s)
-						}
-					})
-				})
-			})
-
-			when("the launch, cache and build flags are false", func() {
-				when("the flags are specified in <layer>.toml", func() {
-					it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
-						h.Mkdir(t,
-							filepath.Join(layersDir, "A", "layer"),
-						)
-						h.Mkfile(t,
-							"[types]\n  build=false\n  cache=false\n  launch=false",
-							filepath.Join(layersDir, "A", "layer.toml"),
-						)
-
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNil(t, err)
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "A", "layer"))
-						h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
-					})
-				})
-
-				when("the flags aren't specified in <layer>.toml", func() {
-					it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
-						h.Mkdir(t,
-							filepath.Join(layersDir, "A", "layer"),
-						)
-						h.Mkfile(t,
-							"",
-							filepath.Join(layersDir, "A", "layer.toml"),
-						)
-
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNil(t, err)
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "A", "layer"))
-						h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
-					})
-				})
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L398-L413